SocketRocket/SRWebSocket.m

```diff
@@ -563,14 +583,27 @@ - (void)_failWithError:(NSError *)error;
     });
 }
 
-- (void)_writeData:(NSData *)data;
+- (void)_writeData:(NSData *)data
 {
-    [self assertOnWorkQueue];
+    [self _writeData:data complete:NULL];
+}
 
+- (void)_writeData:(NSData *)data complete:(SRSendCompleteBlock)complete {
+    [self assertOnWorkQueue];
+    
     if (_closeWhenFinishedWriting) {
+        if (complete) {
+            complete(SRErrorWithCodeDescription(2134, @"socket is closed"));
+        }
         return;
     }
-
+    
+    SRDataCallback *record = [[SRDataCallback alloc] init];
+    record.completeBlock = complete;
+    NSUInteger location = dispatch_data_get_size(_outputBuffer);
+    record.range = NSMakeRange(location, [data length]);
```

Comment:
Nit: Please use dot-syntax for ObjC properties aka `data.length` instead of `[data length]`.